plugins/parsers/csv/parser.go

```diff
@@ -61,6 +62,8 @@ type Parser struct {
 	DefaultTags  map[string]string
 	metadataTags map[string]string
 
+	parseMutex sync.Mutex
+
```

Comment:
How about moving this to the end of the struct and embed the mutex like

```suggestion
	sync.Mutex
```

This way you can simply write `p.Lock()` and `p.Unlock()` and it becomes clear that the intend is to lock the whole parser...